### DISCUSSION TITLE
Router does NOT use custom start point if GPS is on

### DISCUSSION THREAD

**Discussion by 6543** [osmandapp/OsmAnd#11846] [Q&A]
### Description
If you like to calculate/plan a new route for point A to point B but your GPS point to C, router will always use C as startpoint.

### Workaround
Disable GPS nad wait until osm and doesn't recognice your current location anymore - then it works like expected

### How to reproduce?
Enable GPS, try to let OsmAnd~ calculate a route from point A (not your current location) to point B (somewhere else)

### Your Environment
OsmAnd Version: 3.9.10 (F-Droid)
Android Version: 10 
Device model: OnePlus 5T

**Comment by vshcherb** [comment]
This is standard behavior for Navigation. If you are on Route Preparation step, don't click navigate, select start point and view the route. Once you click button Start / Navigate you enter Navigation mode and the route will recalculate or adjust to your current location

**Reply by 6543** [reply]
@vshcherb so it's more a iretating UX ... I think it still would be an **enhancement** since I'm not the only one fall into this trap, some other relatives told me the same issue too ...

what do you think of: stop naviagtion mode when you set a new start point not next to you?

**Reply by vshcherb** [reply]
Well navigation is about following mode, it can't be nothing else as current location otherwise it would be a simulation. I would agree with you if OsmAnd is only about Route planning application but it has 2 different modes with different UX.

**Reply by Lee-Carre** [reply]
Sounds like something else for #7655.

**Reply by barteis** [reply]
I don't recall any GPS--be it a unit or an app--in which "start navigation" does anything but navigate from your current position.

**Reply by 6543** [reply]
I think in this case what about an option "calculate route" ?

**Reply by 6543** [reply]
IDEA: have a "start navigate" who ignore if start point is set & a calculator where only if a start & end is set appear and follow that one?

that way you have two function covered: naviagation & route planing

**Reply by Lee-Carre** [reply]
Perhaps if a route origin which isn't `my location` is selected, then temporarily disable route-recalculation, in order that the navigation instructions are first concerned with getting the user back on the origin-to-destination path?

Other info (from @6543) which I feel would be helpful, here, especially from a usability perspective, is
* what's the goal which you're trying to achieve, which is different from typical (from where-I-am to selected destination) navigation?
* if a user does as your OP describes, but taps Start Navigation, and GNSS position is significantly different to the selected origin, what do you think the app should do in such cases? Maybe you already addressed this, and I did read your earlier postings; I remain unclear on this point
* an important factor to keep in mind is to avoid the situation where a user wants to do regular navigation (from his location to selected destination), but isn't aware that an origin other than his GNSS position is selected or that `my position` *should* be selected; suggestions for this case?
* for the purposes of planning a route, or having OsmAnd follow a specific route, in what ways is the Plan Route feature deficient in meeting your requirements?

I feel like something's missing, in all this. As if we're talking past each other. Often, this suggests a more fundamental misapprehension of intent, or some other miscommunication.

**Reply by tommycrock** [reply]
>  if a start & end is set appear and follow that one?

Another question for @6543 - in what (useful) way would this 'follow' the route if the route is not where you are? Have you found the 'simulate navigation' option vshcherb mentioned in navigation settings? Perhaps this isn't the functionality you want.

**Reply by th1j5** [reply]
Sooo, I'll be reviving this discussion :sweat_smile: 

My dad was stumped by this behaviour, and it made him actually dismiss this app as being 'too buggy to use' (while this is 1 of the best apps there is!).
But even for me (a technical user), it was not clear how I could inspect the proposed route from A to B, without triggering a recalculation.

Just think about the UX for a regular person, wanting to check their route for the next day:
1. You enter A (different from current location)
2. You enter B
3. You press start to inspect the route from close up
4. Step 1. gets completely ignored

Of course, now I know that you should touch the map somewhere without pressing 'start', which will stash away the navigation dialog and enable all the route inspection you want.

But it's about the UX for a regular user!
And I think the best approach to fix this, would be to have 3 buttons,  
**making these 2 different modes explicit (https://github.com/osmandapp/OsmAnd/discussions/11846#discussioncomment-845991):**
- "**Cancel**" (as is now)
- "**Start**" (as is now) , maybe "Start route" or "Start navigation" or ...
- "**Inspect**" / "Inspect route" / "Calculate route" / "Plan route" / "Calculator" / ...
  - I imagine this would be exactly the same as stashing away the navigation dialog by pressing on the map, **but shown explicitly as a choice by a button**
   - If this is not exactly the same, maybe the inspect mode could enable a 'simulation' widget, which increases the discoverability of that option, and maybe similar options...
     - I'm not convinced myself this is necessarily a good idea, if only for the increased complexity.

(and of course, the 'plan route' feature is not useful here, the user just wants a route calculation from A to B for the next day)

Environment: iPad & iPhone, latest version in app-store (4.9.3)

**Reply by th1j5** [reply]
After discussion with my dad, this seems like a better user experience, but there is an alternative:
Only keep 2 buttons, but depending on the start location, differ their action.
- **"Cancel"** button.
- **"Start"/"Inspect" button**.
  - "**Start**": Only if the start location is the current location (gps), then "Start" is shown. This will start navigation.
  - "**Inspect**": If the start location is different from the current location, then "Inspect" is shown. This will only stash away the menu, maybe add a 'simulation' widget, ... (but again, maybe too much code complexity  - but still a good UX)

This is also how google maps does it, there the "inspect" button is "Voorbeeld" (localized for the dutch language, translates to "Example"). You cannot press "start" when you choose a different starting location.
